src/main/java/org/folio/rest/impl/PostOrderLineHelper.java

```diff
@@ -48,26 +46,23 @@
     super(httpClient, okapiHeaders, asyncResultHandler, ctx);
   }
 
-  public CompletableFuture<PoLine> createPoLineWithAutoGeneratedData(PoLine compPOL, boolean updateInventory) {
-    compPOL.setCreated(new Date());
-    compPOL.setCreatedBy(okapiHeaders.get(OKAPI_USERID_HEADER));
+
+  public CompletableFuture<PoLine> createPoLine(PoLine compPOL, boolean updateInventory) {
 
     if (updateInventory) {
       return getProductTypesMap(compPOL)
         .thenCompose(productTypesMap -> getInstanceRecord(compPOL, productTypesMap))
-        .thenCompose(instanceId -> createPoLine(compPOL, instanceId));
+        .thenCompose(instanceId -> {
+          if (instanceId != null) {
+            compPOL.setInstanceId(instanceId);
+          }
+          return createPoLine(compPOL);
+        });
     }
     return createPoLine(compPOL);
   }
 
   private CompletableFuture<PoLine> createPoLine(PoLine compPOL) {
-    return createPoLine(compPOL, null);
```

Comment:
I would probably leave these piece, so that: 
- all PO line creation logic is in one place (createPoLine method), rather than InstanceId set in one place and the rest of data in another place
- make the calling logic cleaner by not requiring manipulations with InstanceId, e.g.:
    ...
    .thenCompose(instanceId -> createPoLine(compPOL, instanceId));
}
return createPoLine(compPOL);